src/components/course/list/student/CourseList.vue

```diff
@@ -78,14 +79,26 @@
                     courses.value = tmpCourses;
                 } else {
                     let response: APIResponse<Course[]>;
-                    response = await courseManagement.getCourses();
+                    response = await courseManagement.getCourses(undefined, undefined, undefined, await getOwnExamRegNames());
                     courses.value = genericResponseHandler.handleResponse(response);
                 }
                 const lecturerIds = new Set(courses.value.map((course) => course.lecturerId));
                 const resp = await userManagement.getUsers(undefined, Array.from(lecturerIds));
                 lecturers.value = (genericResponseHandler.handleResponse(resp) as User_List).lecturers;
             }
 
+            async function getOwnExamRegNames(): Promise<string[]> {
+                const matriculation_management = new MatriculationManagement();
+                const handler = new GenericResponseHandler("examination regulations");
+                const response = await matriculation_management.getOwnMatriculationHistory();
+                const result = handler.handleResponse(response);
+                let returnValue = [] as string[];
+                result.matriculationStatus.forEach((matriculation) => {
+                    returnValue.push(matriculation.fieldOfStudy);
+                });
+                return returnValue;
+            }
```

Comment:
two remarks: 
- we should only add exam regs that we are currently enrolled in, (check getCurrentSemester() from configuration management)
 - `const returnValue = result.matriculationStatus.map((matriculation) => matriculation.fieldOfStudy);` does the same as lines 95 to 98

Reply:
implemented with https://github.com/upb-uc4/ui-web/pull/851/commits/f49e3563b559bc5852a916bc523289d2162c0916 . To check the current semester, I stayed with the foreach, as I do not know if that is possible with map

Reply:
no need to change it now, but `result.matriculationStatus.filter((matriculation) => matriculation.semesters.includes(currentSemester)).map((matriculation) => matriculation.fieldOfStudy);` would be the oneliner for that